React/wbooks/src/app/components/RouteHandler/index.js

```diff
@@ -1,33 +1,43 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import { Route } from 'react-router-dom';
 import { Redirect } from 'react-router';
 
-import Navbar from '../Navbar/index.js'
-import * as localstorageService from '../../../services/localstorageService';
+import Navbar from '../Navbar';
 
 class RouteHandler extends Component {
-  validateAuth = props => {
-    return localstorageService.retrieveUserTokenFromLocalStorage();
-  };
+  isPublicRequest = props => props.location.pathname === '/' || props.location.pathname === '/signup';
 
-  isPublicRequest = props => {
-    return props.location.pathname === '/' || props.location.pathname === '/signup';
-  };
+  render() {
+    const { component: Element, ...rest } = this.props;
+    const render = props =>
```

Comment:
Not here! not in render, you're declaring this function in each render! move as a class instance function!